test/e2e/aks.go

```diff
@@ -225,46 +225,72 @@ func GetAKSKubernetesVersion(ctx context.Context, e2eConfig *clusterctl.E2EConfi
 	settings, err := auth.GetSettingsFromEnvironment()
 	Expect(err).NotTo(HaveOccurred())
 	subscriptionID := settings.GetSubscriptionID()
-	authorizer, err := settings.GetAuthorizer()
+	maxVersion, err := GetWorkingAKSKubernetesVersion(ctx, subscriptionID, location, e2eAKSVersion)
 	Expect(err).NotTo(HaveOccurred())
+
+	return maxVersion, nil
+}
+
+// byClusterOptions returns a set of ListOptions that allows to identify all the objects belonging to a Cluster.
+func byClusterOptions(name, namespace string) []client.ListOption {
+	return []client.ListOption{
+		client.InNamespace(namespace),
+		client.MatchingLabels{
+			clusterv1.ClusterLabelName: name,
+		},
+	}
+}
+
+// GetWorkingAKSKubernetesVersion returns an available Kubernetes version of AKS given a desired semver version, if possible
+// If the desired version is available, we return it;
```

Comment:
thoughts or incorporating something like https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1767/files#diff-35f348a5d567ebeba39b1e40bbab0e1a7f64009cbadff142952f9b9691e94666R137 for "give me the latest version of this Major.Minor release" ? The previous behavior was to get the latest patch for any valid version but now we're just going to stick to that version if it is valid so I think we should also preserve the existing behavior of "latest available", ideally using the same input format as the other tests, ie. "stable-major.minor".